sidechain/primitives/src/traits/mod.rs

```diff
@@ -26,25 +26,42 @@ use sp_core::{blake2_256, Pair, Public, H256};
 use sp_runtime::traits::Member;
 use sp_std::{fmt::Debug, prelude::*};
 
+pub trait Header {
+	/// Identifier for the shards.
+	type ShardIdentifier: Encode + Decode + Hash + Copy + Member;
+
+	/// get block number
```

Comment:
```suggestion
	/// Get block number.
```

same for the ones below :)